Sync with the main repo

hebasto opened the pull request:
Sync with the main repo up to https://github.com/bitcoin/bitcoin/commit/d05be150ca23c0c1d7a6a908a3c43dce181209a8.

hebasto commented:
@icota 

Would you mind looking into Android stuff of this PR?

There are two main points:

1. To fix building of the `qt` package in depends, the [`fix_android_plugin_names.patch`](https://github.com/bitcoin-core/gui-qml/blob/e91fb346ae1db433d39f743c5deeaaac6a7b2d6b/depends/patches/qt/fix_android_plugin_names.patch) was added. What do you think about it?
2. The produced APK packages are still broken. I assume it is related to referencing to QML resources in `AndroidManifest.xml`, but not sure about it. If you have a chance could you check it out?

hebasto commented:
Updated allowed libraries for Windows Guix build.

hebasto commented:
I've updated Guix builds in the OP according to https://gnusha.org/bitcoin-builds/2021-12-22.log:
> Since Qt 5.13 rcc reads `SOURCE_DATE_EPOCH`. Therefore, each build must have a fresh qt package. The easiest way is to `make -C depends clean`.

jarolrod commented:
@hebasto, our guix hashes match: 

```
c6c6b911219afc9d1c50a9c5f064504502b03aa35dd422ff72f794ceb0a5c0b8  guix-build-b7c585cf64dd/output/aarch64-linux-gnu/SHA256SUMS.part
726a8a82ddd68c2883e4657b73d6215fae292929f9d46407b617c1b6cb85072c  guix-build-b7c585cf64dd/output/aarch64-linux-gnu/bitcoin-b7c585cf64dd-aarch64-linux-gnu-debug.tar.gz
4a0b2777fa2029ddb7afa401cacaf58b76dc2456613ce8689b3ad1ab768414f4  guix-build-b7c585cf64dd/output/aarch64-linux-gnu/bitcoin-b7c585cf64dd-aarch64-linux-gnu.tar.gz
b34716e482c9343afe4be9ad298d198970b45205c688bfa3f32e06d9a3607ea6  guix-build-b7c585cf64dd/output/arm-linux-gnueabihf/SHA256SUMS.part
84f249397cc38c661897b84a4aad6fca03a86b6923bf6e7122f889d73b1c8953  guix-build-b7c585cf64dd/output/arm-linux-gnueabihf/bitcoin-b7c585cf64dd-arm-linux-gnueabihf-debug.tar.gz
cad8f056f98056ba9830fdf5b0348ea4d730c57aa8407c25ec7eb66660a2bd1e  guix-build-b7c585cf64dd/output/arm-linux-gnueabihf/bitcoin-b7c585cf64dd-arm-linux-gnueabihf.tar.gz
3bbdae738179fc52a5c6633b763b4f289fe8ea67e932779f5d26aafb696b5c44  guix-build-b7c585cf64dd/output/dist-archive/bitcoin-b7c585cf64dd.tar.gz
83beeb918dd90716cbbc672e9e91bd04f313a4582ce2d033ba35dd893203ad22  guix-build-b7c585cf64dd/output/powerpc64-linux-gnu/SHA256SUMS.part
ece1468498fcc79c809de4de2c8583ecde720f6c20eef864c58f5fb928cde816  guix-build-b7c585cf64dd/output/powerpc64-linux-gnu/bitcoin-b7c585cf64dd-powerpc64-linux-gnu-debug.tar.gz
6c43c07d7801293e2dd46026364529cef23028dcbe049605064bbf046ba87937  guix-build-b7c585cf64dd/output/powerpc64-linux-gnu/bitcoin-b7c585cf64dd-powerpc64-linux-gnu.tar.gz
f02e0be9de9bf20f6dcc7ded915b07008813fbe060be37ccc2a972d2290d2120  guix-build-b7c585cf64dd/output/powerpc64le-linux-gnu/SHA256SUMS.part
35e491f004c8b997f2ee19d4e41f139931f04ed91f4872a0dea03f9b7db8b7ee  guix-build-b7c585cf64dd/output/powerpc64le-linux-gnu/bitcoin-b7c585cf64dd-powerpc64le-linux-gnu-debug.tar.gz
d5fbdca86614950f436281d6fcf35b6a617df796fdb57f9b840943c1cda35c38  guix-build-b7c585cf64dd/output/powerpc64le-linux-gnu/bitcoin-b7c585cf64dd-powerpc64le-linux-gnu.tar.gz
0422876ab19b1c56b2e84e1927f3a3408a5e5e60655081b093ee880a8572cabf  guix-build-b7c585cf64dd/output/riscv64-linux-gnu/SHA256SUMS.part
49ee3b1adcbfbcb834ceced0a70b4e81dd9615a221ca91e4681179a6efd658e1  guix-build-b7c585cf64dd/output/riscv64-linux-gnu/bitcoin-b7c585cf64dd-riscv64-linux-gnu-debug.tar.gz
67abb4008473dc72bdddbe6fac192da0ba95cf6d13a79682c889d4d9d06dae30  guix-build-b7c585cf64dd/output/riscv64-linux-gnu/bitcoin-b7c585cf64dd-riscv64-linux-gnu.tar.gz
e607ff8b465145f4d2e3b9f137733241e59755a3416b6284e49b766ed6f36718  guix-build-b7c585cf64dd/output/x86_64-apple-darwin/SHA256SUMS.part
eb9ec1d10c261c0736981ca856ccd726b96972517cbb11024d754ccf375cc078  guix-build-b7c585cf64dd/output/x86_64-apple-darwin/bitcoin-b7c585cf64dd-osx-unsigned.dmg
9b9fa8e7e267e448baf71910ef46d5e2fca1fd802c10d855738eef7721b303eb  guix-build-b7c585cf64dd/output/x86_64-apple-darwin/bitcoin-b7c585cf64dd-osx-unsigned.tar.gz
c76e1aa3d21d4fa5df0d38a655cb0f2efbb463ce1db377b0a6efc112ed9b8c56  guix-build-b7c585cf64dd/output/x86_64-apple-darwin/bitcoin-b7c585cf64dd-osx64.tar.gz
bd818561ea8e2d8b2eb6338c17b2482c83dc7e47c7d2dad9af121bbfdff73edd  guix-build-b7c585cf64dd/output/x86_64-linux-gnu/SHA256SUMS.part
90eb86f5a04060e86f5b9052458c9e0980a1ce5a09be9dc9ca69ccf2dc9420a2  guix-build-b7c585cf64dd/output/x86_64-linux-gnu/bitcoin-b7c585cf64dd-x86_64-linux-gnu-debug.tar.gz
5b0f5992f2567ce14659924bd0525bf1856e00db618498685d8aaa74704e7f70  guix-build-b7c585cf64dd/output/x86_64-linux-gnu/bitcoin-b7c585cf64dd-x86_64-linux-gnu.tar.gz
56686c684e3d8173384f19b76fca9077e15cab248473b8a1e1c260e93f9d0c72  guix-build-b7c585cf64dd/output/x86_64-w64-mingw32/SHA256SUMS.part
9aeb4ee4cd3b153e4e8041a4be9ac24af9c6220737f538213f374b897864253e  guix-build-b7c585cf64dd/output/x86_64-w64-mingw32/bitcoin-b7c585cf64dd-win-unsigned.tar.gz
361bde3ed0f227fa761af7af78cf05e089fe708b33a7c1a7f835a8cc931e5fb6  guix-build-b7c585cf64dd/output/x86_64-w64-mingw32/bitcoin-b7c585cf64dd-win64-debug.zip
d014c2859afae54d39db8ef3a67ffdf24cfcb125dc00166d0854927647486e79  guix-build-b7c585cf64dd/output/x86_64-w64-mingw32/bitcoin-b7c585cf64dd-win64-setup-unsigned.exe
f852887edd3c06f24764bcec093d7d44d98e2248c14a67cf9f09cb217a50a606  guix-build-b7c585cf64dd/output/x86_64-w64-mingw32/bitcoin-b7c585cf64dd-win64.zip
```

I am currently in the process of repeating the sync and verifying for correctness. Will take me a few days probably.

icota commented:
> 1. To fix building of the `qt` package in depends, the [`fix_android_plugin_names.patch`](https://github.com/bitcoin-core/gui-qml/blob/e91fb346ae1db433d39f743c5deeaaac6a7b2d6b/depends/patches/qt/fix_android_plugin_names.patch) was added. What do you think about it?

I think it's okay. It fixes the problem at hand and it's not like we don't have such workarounds elswhere.

> 2. The produced APK packages are still broken. I guess it is related to referencing to QML resources in `AndroidManifest.xml`, but not sure about it. If you have a chance could you check it out?

It just needed a re-do of https://github.com/bitcoin/bitcoin/commit/6fc5c772d47cca01968c451f070f997c9f7641e0 in the `qml` folder. I made a [PR](https://github.com/hebasto/gui-qml/pull/1/files) to this branch.

hebasto commented:
Updated b7c585cf64dd7f5599b7a0baa62fba37e1c96107 -> a555e19b7c3b49437660236b4cd2e32118d18c96 ([pr91.03](https://github.com/hebasto/gui-qml/commits/pr91.03) -> [pr91.04](https://github.com/hebasto/gui-qml/commits/pr91.04)):

- rebased
- included more recent commits from the main repo, and another merge conflict with bitcoin/bitcoin#23845 resolved
- b1b630476056159873134341885d960fe2856518 processed to address @icota's [comment](https://github.com/bitcoin-core/gui-qml/pull/91#issuecomment-1000544165)

hebasto commented:
Updated b7c585cf64dd7f5599b7a0baa62fba37e1c96107 -> c8f4e0398d0e45c0aaef0d98b7835c16080a07ea ([pr91.03](https://github.com/hebasto/gui-qml/commits/pr91.03) -> [pr91.05](https://github.com/hebasto/gui-qml/commits/pr91.05)):

- rebased
- included more recent commits from the main repo, and another merge conflict with bitcoin/bitcoin#23845 resolved
- b1b630476056159873134341885d960fe2856518 processed to address @icota's [comment](https://github.com/bitcoin-core/gui-qml/pull/91#issuecomment-1000544165)

hebasto commented:
Guix build hashes in the OP have been updated.

jarolrod commented:
new hashes, mine match @hebasto:

```
69492cd3777954f2771cfe8e733c8dba18a9182310a3fbd16b04e3663c846e46  guix-build-c8f4e0398d0e/output/aarch64-linux-gnu/SHA256SUMS.part
b2b3a06024980b83e9c8277fdb0f70ec7d5d15ada04124e13d1746baf5c6ba95  guix-build-c8f4e0398d0e/output/aarch64-linux-gnu/bitcoin-c8f4e0398d0e-aarch64-linux-gnu-debug.tar.gz
cc0404ba6c5161a7da72fe1905d4476209ad2e43796df3d667b844c74f313f2e  guix-build-c8f4e0398d0e/output/aarch64-linux-gnu/bitcoin-c8f4e0398d0e-aarch64-linux-gnu.tar.gz
c14a4f1cee523117be00863b5751c03b9ca24cddf76920adf28cf29adef5016c  guix-build-c8f4e0398d0e/output/arm-linux-gnueabihf/SHA256SUMS.part
f1d0b7202223296bcdde9878c9bfd60253c2d2530a74a3c94ed82b90568cc866  guix-build-c8f4e0398d0e/output/arm-linux-gnueabihf/bitcoin-c8f4e0398d0e-arm-linux-gnueabihf-debug.tar.gz
d07722b96f07166328d218716e3ff7dabcf8f6d35f906378214a97b88f6cbb09  guix-build-c8f4e0398d0e/output/arm-linux-gnueabihf/bitcoin-c8f4e0398d0e-arm-linux-gnueabihf.tar.gz
89ba246d916d5805159bd92d23bd620a3cf6c6bdee962214120c5005eabd4b75  guix-build-c8f4e0398d0e/output/dist-archive/bitcoin-c8f4e0398d0e.tar.gz
1dcf2101a3c766724d0fdbeca7c61b3d501081d3c166b3b9d9e52418f6f34bb0  guix-build-c8f4e0398d0e/output/powerpc64-linux-gnu/SHA256SUMS.part
b563412fc17fc5189a8e8320096113f4d5c42011904c78e92db21929e8c565f5  guix-build-c8f4e0398d0e/output/powerpc64-linux-gnu/bitcoin-c8f4e0398d0e-powerpc64-linux-gnu-debug.tar.gz
fa38f1520dc04e2fca68031bbd13ff67c42c7bf26003e93a7927dde8e904da6b  guix-build-c8f4e0398d0e/output/powerpc64-linux-gnu/bitcoin-c8f4e0398d0e-powerpc64-linux-gnu.tar.gz
beac355a65c78c93963c23707667ce7a96e2550edcdad37dfc7c1ce6dbc5bf14  guix-build-c8f4e0398d0e/output/powerpc64le-linux-gnu/SHA256SUMS.part
c813ba3b3de89d331138cdcec609efe0fd7e3320b3c28a7c4fcf5537bb05c614  guix-build-c8f4e0398d0e/output/powerpc64le-linux-gnu/bitcoin-c8f4e0398d0e-powerpc64le-linux-gnu-debug.tar.gz
2d22178891c3a605a1a4e2a35797bf4e38dfea0245189605d7baf7165227ce75  guix-build-c8f4e0398d0e/output/powerpc64le-linux-gnu/bitcoin-c8f4e0398d0e-powerpc64le-linux-gnu.tar.gz
9290ceb49f237f78ae0aa1ff746f4a36d0cfe0db2caf82811c263133b4dab9d6  guix-build-c8f4e0398d0e/output/riscv64-linux-gnu/SHA256SUMS.part
562ddb5af2cd8d605bb933d516210525459398f52e7a6070e930c93fd2861b80  guix-build-c8f4e0398d0e/output/riscv64-linux-gnu/bitcoin-c8f4e0398d0e-riscv64-linux-gnu-debug.tar.gz
65c6faa6a69e288ff1cf26d2f57dc407ef398e6eb86aac9f38035037960b4472  guix-build-c8f4e0398d0e/output/riscv64-linux-gnu/bitcoin-c8f4e0398d0e-riscv64-linux-gnu.tar.gz
647c127b1b656e667bcb4435a4472ee680027515f6cecd7b0a07f022dc7cf5e3  guix-build-c8f4e0398d0e/output/x86_64-apple-darwin/SHA256SUMS.part
3eeb8449b9cb350ff442211df7b4f6858fbff610ed4157d1585800efc16bab81  guix-build-c8f4e0398d0e/output/x86_64-apple-darwin/bitcoin-c8f4e0398d0e-osx-unsigned.dmg
b5f9b759fb69a349f41557d0f78df2367f7d52f56d3c2e8f71ee4533b5be4bc7  guix-build-c8f4e0398d0e/output/x86_64-apple-darwin/bitcoin-c8f4e0398d0e-osx-unsigned.tar.gz
83b98e6c298dbf32997d90d1001d8257eeffec51a3feea7df454d122b5d512a5  guix-build-c8f4e0398d0e/output/x86_64-apple-darwin/bitcoin-c8f4e0398d0e-osx64.tar.gz
ad7a69cba6a331e36a7069f2391903b4f63584eccbd74009dffc2dd135d18efe  guix-build-c8f4e0398d0e/output/x86_64-linux-gnu/SHA256SUMS.part
290ace50b1049a8494370a68191b86e4beab15b0afe0947ea2090f98bc1c037c  guix-build-c8f4e0398d0e/output/x86_64-linux-gnu/bitcoin-c8f4e0398d0e-x86_64-linux-gnu-debug.tar.gz
6713f955a9d4ab488cfaf3b97d4d59a0a11c6bdaeba27ddbb15a2d1fb82e3c7c  guix-build-c8f4e0398d0e/output/x86_64-linux-gnu/bitcoin-c8f4e0398d0e-x86_64-linux-gnu.tar.gz
f85f19a7437a3eec7597cb6e855d6e529051634bdea74b94e2d17fb1862d810e  guix-build-c8f4e0398d0e/output/x86_64-w64-mingw32/SHA256SUMS.part
d307ecae13388ed373d8252971d5b9949a21e016d03bd2726a4455a7af35f8ba  guix-build-c8f4e0398d0e/output/x86_64-w64-mingw32/bitcoin-c8f4e0398d0e-win-unsigned.tar.gz
1f281e0e42cf8350ec9d51c875b8fff7f20b11e5e33dda08dcb2ee8a2c33615a  guix-build-c8f4e0398d0e/output/x86_64-w64-mingw32/bitcoin-c8f4e0398d0e-win64-debug.zip
a8c04d2a61852ed9bbd9a01ed9419c3eb52c788d942a5b24f8e45c2d29d0df69  guix-build-c8f4e0398d0e/output/x86_64-w64-mingw32/bitcoin-c8f4e0398d0e-win64-setup-unsigned.exe
a67317e2ef08346858aa766176a79449639c1780ed4499d7a14e2ebcf7ab9ffc  guix-build-c8f4e0398d0e/output/x86_64-w64-mingw32/bitcoin-c8f4e0398d0e-win64.zip
```

promag commented:
Tested ACK c8f4e0398d0e45c0aaef0d98b7835c16080a07ea on macOS 12.0.1, Android 10, and Windows 10. Guix build hashes match as well.
